packages/jest-mock/src/index.ts

```diff
@@ -137,43 +139,58 @@ export interface MockInstance<T, Y extends Array<unknown>> {
 
 type Unpromisify<T> = T extends Promise<infer R> ? R : never;
 
-/**
- * Possible types of a MockFunctionResult.
- * 'return': The call completed by returning normally.
- * 'throw': The call completed by throwing a value.
- * 'incomplete': The call has not completed yet. This is possible if you read
- *               the  mock function result from within the mock function itself
- *               (or a function called by the mock function).
- */
-type MockFunctionResultType = 'return' | 'throw' | 'incomplete';
-
-/**
- * Represents the result of a single call to a mock function.
- */
-type MockFunctionResult = {
+type MockFunctionResultIncomplete = {
+  type: 'incomplete';
   /**
-   * Indicates how the call completed.
+   * Result of a single call to a mock function that has not yet completed.
+   * This occurs if you test the result from within the mock function itself,
+   * or from within a function that was called by the mock.
    */
-  type: MockFunctionResultType;
+  value: undefined;
+};
+type MockFunctionResultReturn<T> = {
+  type: 'return';
   /**
-   * The value that was either thrown or returned by the function.
-   * Undefined when type === 'incomplete'.
+   * Result of a single call to a mock function that returned.
+   */
+  value: T;
```

Comment:
Might be I misunderstood something, but if these TSDoc comments are placed above `type`, they don’t show up in IDE. That’s how they are placed in `@types/jest`. So I was trying to find some place where they could be useful.